### ISSUE TITLE
Compose 1.6.0 discoloured bug(wrong colors) Desktop

### ISSUE THREAD

**Issue by vladyslavv-ua** [JetBrains/compose-multiplatform#4397]
**Describe the bug**
After update org.jetbrains.compose plugin to 1.6.0 something with colors went wrong

**Affected platforms**

- [ ] Desktop (Windows, Linux, macOS)

**Versions**
- Kotlin version*: 1.9.22
- Compose Multiplatform version*: 1.6.0
- OS version(s)* (required for Desktop and iOS issues): Windows 11 Pro
- OS architecture (x86 or arm64): x64
- JDK (for desktop issues): 17.0.2

**To Reproduce**
Steps and/or the code snippet to reproduce the behavior:
1. Go to https://github.com/vladyslavv-ua/compose-desktop-discoloured-bug
2. Download project
3. Run gradle task "compose desktop -> run"

**Expected behavior**
Correct colors

**Screenshots**
compose 1.6.0
![image](https://github.com/JetBrains/compose-multiplatform/assets/110237787/245295a7-30b9-4aee-8939-a519d58dd1d0)

compose 1.5.12
![image](https://github.com/JetBrains/compose-multiplatform/assets/110237787/4cc58f24-5265-4aba-aa68-27bcc9955c79)

**Comment by igordmn** [comment]
Thanks!

We tested it on Windows 11 Home 23H2, it works fine on DIRECT3D/OpenGL/Software renderers. To solve this issue, we need your  help.

Could you tell what will be printed if you run this code?
```
import androidx.compose.runtime.Composable
import androidx.compose.ui.window.Window
import androidx.compose.ui.window.application

fun main() = application {
    Window(onCloseRequest = ::exitApplication, title = "KotlinProject") {
        println(window.renderApi)
        App()
    }
}
```

Also, could you send a screenshot with information from `Settings -> System -> Display -> Advanced display` and tell which GPU driver version do you have?

**Comment by vladyslavv-ua** [comment]
Output of println(window.renderApi) is DIRECT3D

driver is AMD Radeon Graphics v30.0.13002.19003

**Comment by vladyslavv-ua** [comment]
![image](https://github.com/JetBrains/compose-multiplatform/assets/110237787/f6d95b69-f5d4-46d2-adcf-ad0482652465)

**Comment by igordmn** [comment]
Looks usual 🤔. We'll try different systems.

It is a complete shot in the dark, but to be sure that it is DIRECT3D-only issue, worth to try this:
```
import androidx.compose.ui.window.Window
import androidx.compose.ui.window.application

fun main() {
    System.setProperty("skiko.renderApi", "OPENGL")
    // System.setProperty("skiko.renderApi", "SOFTWARE") // or this
    application {
        Window(onCloseRequest = ::exitApplication, title = "KotlinProject") {
            App()
        }
    }
}
```

**Comment by vladyslavv-ua** [comment]
Result of System.setProperty("skiko.renderApi", "OPENGL") and System.setProperty("skiko.renderApi", "SOFTWARE")
![image](https://github.com/JetBrains/compose-multiplatform/assets/110237787/1b08969d-4560-44b5-9543-6f9bde2d866e)

**Comment by igordmn** [comment]
Thanks!

So it is a Direct3D issue.

The are some commits in Skiko that can cause this issue.

Worth to try different versions to identify where the regression is introduced:
0.7.90 // contains https://github.com/JetBrains/skiko/commit/5a29cb59b1e5990a7f33d693e92a8b82de1d53de
0.7.89
0.7.86 // contains https://github.com/JetBrains/skiko/commit/4217f641562b060cca60be61bccaaf9b05cececc
0.7.85.4

in `build.gradle.kts` in the end of file:
```
configurations.all {
    resolutionStrategy.eachDependency {
        if (requested.group.startsWith("org.jetbrains.skiko")) {
            useVersion("0.7.89")
        }
    }
}
```

**Comment by vladyslavv-ua** [comment]
0.7.85.4 Good, 
0.7.86 Good,
0.7.89 Good,
0.7.90 Bad

**Comment by igordmn** [comment]
Thanks! Most probably it is https://github.com/JetBrains/skiko/commit/5a29cb59b1e5990a7f33d693e92a8b82de1d53de, we'll investigate what can cause this.

**Comment by vladyslavv-ua** [comment]
Thank you.

**Comment by MatkovIvan** [comment]
Hi @vladyslavv-ua,

I've added a fallback to old behaviour (without transparency support) here - https://github.com/JetBrains/skiko/pull/856
This fix is available in skiko `0.7.95`

Since I wasn't able to reproduce this locally, I'm not sure that it will be enough for your problem. In the case if `0.7.95` is not work, there is another more aggressive approach where old behaviour is always applied if transparency is not required at the moment of creation. I would avoid applying that if possible, but just in case, I published it as `0.7.95.1`.

Could you please try these versions and report back if it works?

**Comment by vladyslavv-ua** [comment]
Very interesting
skiko 0.7.95
![image](https://github.com/JetBrains/compose-multiplatform/assets/110237787/1c260c5b-413f-4e73-8e3e-25f3c1b820c4)

skiko 0.7.95.1
![image](https://github.com/JetBrains/compose-multiplatform/assets/110237787/bdd57862-0f83-4e90-b93a-c5e587a4c293)

**Comment by vladyslavv-ua** [comment]
By the way when I resize window it fixes a bit

https://github.com/JetBrains/compose-multiplatform/assets/110237787/a0b0cbe5-5efa-4ddc-8445-051fe22157c2

**Comment by MatkovIvan** [comment]
We'll include the fix from `0.7.95.1` to Compose `1.6.1`. I'm not closing this issue because it's still a case for transparent view/window.

@vladyslavv-ua it's interesting resizing behavior, thanks for posting it.

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-4397">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
